client/pages/Event/EventsPage.jsx

```diff
@@ -76,13 +76,20 @@ class EventsPage extends React.Component {
     return null;
   }
 
-  getFilteredEvents({ startDate, location, category }) {
+  getFilteredEvents({ startDate, filterLocation, filterCategory }) {
+    const {
+      selectedVenue,
+      selectedCategory,
+    } = this.state;
+    const location = filterLocation === null ? null : (filterLocation || selectedVenue);
```

Comment:
Nice making the variable names more specific. However, I don't think the implementation here needs to be modified this way. With this, when filterLocation or filterCategory is null, this filter.selectedVenue and filter.selectedCategory is set to null but this makes `Object.keys(filter).length` evaluate to a truthy value and the filter object is returned. This then results in an unnecessary re-render. 
(filterLocation || selectedVenue) will not be needed because selectedVenue will remain in state and we won't need to set it again.

If there is anything I am missing kindly share with me in the reply. Thank you.

Reply:
Thanks for pointing this out @Alameen688 but the reason this was done this way is that when you choose All in any of the filters, the id that will be returned will be null, and it has to be null because that is the only way to reset the filters to neutral. In a case where a filter is not picked at all, say I only choose to filter by category and I didn't pick any location to filter by, then that particular filter parameter will be returned as undefined, when this is done the location will then fall back to the initial selected location on the events page. I hope this clarifies.